consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/PostMergeContext.java

```diff
@@ -73,35 +71,38 @@ public static synchronized PostMergeContext get() {
 
   @Override
   public PostMergeContext setTerminalTotalDifficulty(final Difficulty newTerminalTotalDifficulty) {
+    if (newTerminalTotalDifficulty == null) {
+      throw new IllegalStateException("cannot set null terminal total difficulty");
+    }
     terminalTotalDifficulty.set(newTerminalTotalDifficulty);
     return this;
   }
 
   @Override
   public void setIsPostMerge(final Difficulty totalDifficulty) {
-    if (Optional.ofNullable(lastFinalized.get()).isPresent()) {
+    if (isPostMerge.get().orElse(Boolean.FALSE)
```

Comment:
my understanding is that if we have a finalized block, then we are post merge, so this check seems redundant, am I missing something?

Reply:
Yes the finalized check is unnecessary now.  The post-merge condition is 'stronger' - we will never switch back to the prior consensus mechanism and will instead rely on forkchoiceUpdated to direct any reorg around TTD.  

I will remove the unnecessary finalized check and update the comment 👍